shell.nix for Buildroot is missing flock

Reproduce with
```
git clone https://gitlab.com/buildroot.org/buildroot.git
cp nix-environments/envs/buildroot/shell.nix buildroot/
cd buildroot
nix-shell --pure
make O=build_kvm_x86-64_nix qemu_x86_64_defconfig
make O=build_kvm_x86-64_nix -j8
```

Output
```
<omitting nix-shell --pure building a bunch of things successfully>
<omitting defconfig running a bunch of things>
..
mkdir -p /home/user/dev/git/buildroot/build_kvm_x86-64_nix/build/buildroot-config/lxdialog
PKG_CONFIG_PATH="/usr/lib/pkgconfig" make CC="/usr/bin/gcc" HOSTCC="/usr/bin/gcc" \
    obj=/home/user/dev/git/buildroot/build_kvm_x86-64_nix/build/buildroot-config -C support/kconfig -f Makefile.br conf
/usr/bin/gcc -DCURSES_LOC="<curses.h>" -DLOCALE  -I/home/user/dev/git/buildroot/build_kvm_x86-64_nix/build/buildroot-config -DCONFIG_=\"\"  -MM *.c > /home/user/dev/git/buildroot/build_kvm_x86-64_nix/build/buildroot-config/.depend 2>/dev/null || :
/usr/bin/gcc -DCURSES_LOC="<curses.h>" -DLOCALE  -I/home/user/dev/git/buildroot/build_kvm_x86-64_nix/build/buildroot-config -DCONFIG_=\"\"   -c conf.c -o /home/user/dev/git/buildroot/build_kvm_x86-64_nix/build/buildroot-config/conf.o
/usr/bin/gcc -DCURSES_LOC="<curses.h>" -DLOCALE  -I/home/user/dev/git/buildroot/build_kvm_x86-64_nix/build/buildroot-config -DCONFIG_=\"\"  -I. -c /home/user/dev/git/buildroot/build_kvm_x86-64_nix/build/buildroot-config/zconf.tab.c -o /home/user/dev/git/buildroot/build_kvm_x86-64_nix/build/buildroot-config/zconf.tab.o
/usr/bin/gcc -DCURSES_LOC="<curses.h>" -DLOCALE  -I/home/user/dev/git/buildroot/build_kvm_x86-64_nix/build/buildroot-config -DCONFIG_=\"\"   /home/user/dev/git/buildroot/build_kvm_x86-64_nix/build/buildroot-config/conf.o /home/user/dev/git/buildroot/build_kvm_x86-64_nix/build/buildroot-config/zconf.tab.o  -o /home/user/dev/git/buildroot/build_kvm_x86-64_nix/build/buildroot-config/conf
rm /home/user/dev/git/buildroot/build_kvm_x86-64_nix/build/buildroot-config/zconf.tab.c
  GEN     /home/user/dev/git/buildroot/build_kvm_x86-64_nix/Makefile
#
# configuration written to /home/user/dev/git/buildroot/build_kvm_x86-64_nix/.config
#
(ins)buildroot-fhsenv:user@chimera:~/dev/git/buildroot$ make O=build_kvm_x86-64_nix -j8
/usr/bin/make -j1  O=/home/user/dev/git/buildroot/build_kvm_x86-64_nix HOSTCC="/usr/bin/gcc" HOSTCXX="/usr/bin/g++" syncconfig
make[2]: warning: -j1 forced in submake: resetting jobserver mode.
  GEN     /home/user/dev/git/buildroot/build_kvm_x86-64_nix/Makefile
>>> host-skeleton  Extracting
>>> host-skeleton  Patching
>>> host-skeleton  Configuring
>>> host-skeleton  Building
>>> host-skeleton  Installing to host directory
# check-package DoNotInstallToHostdirUsr
>>> host-pkgconf 2.3.0 Downloading
/usr/bin/sh: line 1: flock: command not found
make[1]: *** [package/pkg-generic.mk:179: /home/user/dev/git/buildroot/build_kvm_x86-64_nix/build/host-pkgconf-2.3.0/.stamp_downloaded] Error 127
make: *** [Makefile:83: _all] Error 2
```

--- a/envs/buildroot/shell.nix
+++ b/envs/buildroot/shell.nix
@@ -56,6 +56,7 @@ in
       expect # not mentioned in buildroot deps
       file
       findutils
+      flock
       gcc
       glib # not mentioned; not sure if necessary
       glibc # transitively mentioned: debian build-essential
@@ -75,7 +76,7 @@ in
     ] ++ pkgs.linux.nativeBuildInputs ++ extraPkgs);
     runScript = ''
       # The host-uboot-tools package uses objcopy from the shells OBJCOPY var
-      # Since the var is set to OBJCOPY=objcopy the buildroot provided 
+      # Since the var is set to OBJCOPY=objcopy the buildroot provided
       # CROSS_COMPILE path is ignored hence the script is using the wrong objcopy
       unset $OBJCOPY
       exec bash

